setup_build.py

```diff
@@ -134,38 +151,29 @@ def run(self):
 
         defs_file = localpath('h5py', 'defs.pyx')
         func_file = localpath('h5py', 'api_functions.txt')
-        config_file = localpath('h5py', 'config.pxi')
 
         # Rebuild low-level defs if missing or stale
         if not op.isfile(defs_file) or os.stat(func_file).st_mtime > os.stat(defs_file).st_mtime:
             print("Executing api_gen rebuild of defs")
             api_gen.run()
 
         # Rewrite config.pxi file if needed
-        s = """\
+        s = f"""\
 # This file is automatically generated by the h5py setup script.  Don't modify.
 
-DEF MPI = %(mpi)s
-DEF ROS3 = %(ros3)s
-DEF DIRECT_VFD = %(direct_vfd)s
-DEF HDF5_VERSION = %(version)s
+DEF MPI = {bool(config.mpi)}
+DEF ROS3 = {bool(config.ros3)}
+DEF HDF5_VERSION = {config.hdf5_version}
+DEF DIRECT_VFD = {bool(config.direct_vfd)}
 DEF SWMR_MIN_HDF5_VERSION = (1,9,178)
 DEF VDS_MIN_HDF5_VERSION = (1,9,233)
 DEF VOL_MIN_HDF5_VERSION = (1,11,5)
-DEF COMPLEX256_SUPPORT = %(complex256_support)s
-DEF NUMPY_BUILD_VERSION = '%(numpy_version)s'
-DEF CYTHON_BUILD_VERSION = '%(cython_version)s'
+DEF COMPLEX256_SUPPORT = {complex256_support}
+DEF NUMPY_BUILD_VERSION = '{numpy.__version__}'
+DEF CYTHON_BUILD_VERSION = '{cython_version}'
 """
-        s %= {
-            'mpi': bool(config.mpi),
-            'ros3': bool(config.ros3),
-            'direct_vfd': bool(config.direct_vfd),
-            'version': config.hdf5_version,
-            'complex256_support': hasattr(numpy, 'complex256'),
-            'numpy_version': numpy.__version__,
-            'cython_version': cython_version,
-        }
-        write_if_changed(config_file, s)
+        write_if_changed(localpath('h5py', 'config.pxi'), s)
+        write_if_changed(localpath('h5py', 'config.pxi', root=build_path), s)
```

Comment:
Why do we now need to write it in the build folder as well, when we didn't before? I'm not saying it's wrong, I'd just like to understand it a bit.

Reply:
So... we don't actually need this to make it build on the CI. I had added this earlier when I thought the config file may not have been copied, but the issue was something else.

The reason I left it, is that I think my changes is the correct approach. I had seen issues in two other [projects](https://github.com/kivy/kivy/pull/2934) I contributed to caused by the build-time generated script not being correctly copied or copied too early before it had the correct values. So I figured it's better to be safe than sorry and left the change there. But I can remove it if needed.

Reply:
Thanks! I'd say that if we can't identify a concrete problem with the code as it was before, let's leave it alone and not add to it (i.e. roll back these changes).

At some point, we're probably going to need to overhaul how h5py is built. I'm hoping that we can draw on either [scipy's work to use meson](https://github.com/rgommers/scipy/issues/22), or the [scikit-build overhaul](https://iscinumpy.dev/post/scikit-build-proposal/) for this before our setuptools based build breaks too badly. Until then, I'd like a minimum of changes and a minimum of complexity. :slightly_smiling_face:

Reply:
Ok, done!